pkg/skaffold/jib/jib_init.go

```diff
@@ -0,0 +1,165 @@
+	}
+	return fmt.Sprintf("%s (%s)", j.BuilderName, j.FilePath)
+}
+
+// CreateArtifact creates an Artifact to be included in the generated Build Config
+func (j Jib) CreateArtifact(manifestImage string) *latest.Artifact {
+	workspace := filepath.Dir(j.FilePath)
+
+	a := &latest.Artifact{ImageName: j.Image}
+	if j.Image == "" {
+		a.ImageName = manifestImage
+	}
+
+	if workspace != "." {
+		a.Workspace = workspace
+	}
+
+	if j.BuilderName == JibMaven {
+		jibMaven := &latest.JibMavenArtifact{}
+		if j.Project != "" {
+			jibMaven.Module = j.Project
+		}
+		if j.Image == "" {
+			jibMaven.Flags = []string{"-Dimage=" + manifestImage}
+		}
+		a.ArtifactType = latest.ArtifactType{JibMavenArtifact: jibMaven}
+
+	} else if j.BuilderName == JibGradle {
+		jibGradle := &latest.JibGradleArtifact{}
+		if j.Project != "" {
+			jibGradle.Project = j.Project
+		}
+		if j.Image == "" {
+			jibGradle.Flags = []string{"-Dimage=" + manifestImage}
+		}
+		a.ArtifactType = latest.ArtifactType{JibGradleArtifact: jibGradle}
+	}
+
+	return a
+}
+
+// ConfiguredImage returns the target image configured by the builder, or empty string if no image is configured
+func (j Jib) ConfiguredImage() string {
+	return j.Image
+}
+
+// Path returns the path to the build definition
+func (j Jib) Path() string {
+	return j.FilePath
+}
+
+// BuilderConfig contains information about inferred Jib configurations
+type jibJSON struct {
+	Image   string `json:"image"`
+	Project string `json:"project"`
+}
+
+// ValidateJibConfig checks if a file is a valid Jib configuration. Returns the list of Config objects corresponding to each Jib project built by the file, or nil if Jib is not configured.
+func ValidateJibConfig(path string) []Jib {
+	// Determine whether maven or gradle
+	var builderType, executable, wrapper, taskName string
+	switch {
+	case strings.HasSuffix(path, "pom.xml"):
+		builderType = JibMaven
+		executable = "mvn"
+		wrapper = "mvnw"
+		taskName = "jib:_skaffold-init"
+	case strings.HasSuffix(path, "build.gradle"):
+		builderType = JibGradle
+		executable = "gradle"
+		wrapper = "gradlew"
+		taskName = "_jibSkaffoldInit"
+	default:
+		return nil
+	}
+
+	// Run Jib's skaffold init task/goal to check if Jib is configured
+	if wrapperExecutable, err := util.AbsFile(filepath.Dir(path), wrapper); err == nil {
+		executable = wrapperExecutable
+	}
+	cmd := exec.Command(executable, taskName, "-q")
+	cmd.Dir = filepath.Dir(path)
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return nil
+	}
+
+	// Parse Jib output. Multiple JSON strings may be printed for multi-project/multi-module setups.
+	matches := regexp.MustCompile(`BEGIN JIB JSON\r?\n({.*})`).FindAllSubmatch(stdout, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+
+	results := make([]Jib, len(matches))
+	for i, match := range matches {
+		line := bytes.Replace(match[1], []byte(`\`), []byte(`\\`), -1)
+		parsedJSON := jibJSON{}
+		if err := json.Unmarshal(line, &parsedJSON); err != nil {
+			return nil
```

Comment:
This should usually not happen, but may happen if we change the JSON structure? Should we log this or print a warning?